android/src/main/java/com/gr4vy/embedreactnative/EmbedReactNativeModule.java

```diff
@@ -61,6 +65,25 @@ public static <T> T coalesce(T... items) {
     return null;
   }
 
+  public static Object getMapOptionalValue(ReadableMap map, String key) {
+    if (map.hasKey(key)) {
+      if (!map.isNull(key)) {
+        switch (map.getType(key)) {
+          case Number:
```

Comment:
[Number](https://docs.oracle.com/javase/8/docs/api/java/lang/Number.html) includes other concrete types also exposed in `ReadableMap` like `Integer`, `Long` and `Double`. Should we check for those types instead?

Reply:
as discussed, guess we're stuck with the RN custom `Number` https://github.com/facebook/react-native/blob/main/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/ReadableType.kt